packages/gestalt/src/Masonry/defaultMultiColumnLayout.js

```diff
@@ -0,0 +1,67 @@
+// @flow strict
+import { type Cache } from './Cache';
+import multiColumnLayout from './multiColumnLayout';
+import { type Position } from './types';
+
+const offscreen = (width: number, height: number = Infinity) => ({
+  top: -9999,
+  left: -9999,
+  width,
+  height,
+});
+
+const defaultLayout =
+  <T: { +[string]: mixed }>({
+    columnWidth = 236,
+    gutter = 14,
+    justify,
+    minCols = 2,
+    rawItemCount,
+    width,
+    measurementCache,
+    ...otherProps
+  }: {
+    columnWidth?: number,
+    gutter?: number,
+    justify: 'center' | 'start',
+    minCols?: number,
+    rawItemCount: number,
+    width?: ?number,
+    logWhitespace?: (number) => void,
+    measurementCache: Cache<T, number>,
+    positionCache: Cache<T, Position>,
+    whitespaceThreshold?: number,
+  }): ((items: $ReadOnlyArray<T>) => $ReadOnlyArray<Position>) =>
+  (items: $ReadOnlyArray<T>): $ReadOnlyArray<Position> => {
+    if (width == null) {
+      return items.map(() => offscreen(columnWidth));
+    }
+
+    const columnWidthAndGutter = columnWidth + gutter;
+    const columnCount = Math.max(Math.floor((width + gutter) / columnWidthAndGutter), minCols);
+
+    let centerOffset;
+    if (justify === 'center') {
+      const contentWidth = Math.min(rawItemCount, columnCount) * columnWidthAndGutter + gutter;
+
+      centerOffset = Math.max(Math.floor((width - contentWidth) / 2), 0);
+    } else {
```

Comment:
same comment/question here about just moving this into the base layout function

totally fine if you'd rather be more cautious initially